homeworks/10/Chat-room/src/Server.java

```diff
@@ -0,0 +1,152 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class Server implements Runnable{
+
+    private ArrayList<ConnectionHandler> connectionHandlers;
+    private ServerSocket server;
+    private boolean done;
+
+    private ExecutorService pool;
+
+    public Server(){
+        connectionHandlers = new ArrayList<>();
+        done = false;
+    }
+
+    @Override
+    public void run() {
+        try {
+            server = new ServerSocket(1234);
+            pool = Executors.newCachedThreadPool();
+            while(!done) {
+                Socket client = server.accept();
+                ConnectionHandler handler = new ConnectionHandler(client);
+                connectionHandlers.add(handler);
+                pool.execute(handler);
+            }
+        } catch (IOException e){
+            close();
+        }
+
+    }
+
+    public void broadcastMessage(String message){
+        for(ConnectionHandler conn: connectionHandlers){
+            if (message!= null){
+                if(conn.sender == true){
+                    continue;
+                }
+                conn.sendMessage(message);
+            }
+
+        }
+    }
+
+    public void close(){
+        done = true;
+        if (!server.isClosed()){
+            try {
+                server.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        for(ConnectionHandler ch: connectionHandlers){
+            ch.close();
+        }
+    }
+
+
+    class ConnectionHandler implements Runnable{
+
+        private Socket client;
+        private BufferedReader in;
+        private PrintWriter out;
+        private String nickname;
+        private boolean sender;
+
+        public ConnectionHandler(Socket client) {
+            this.client = client;
+            sender = false;
+        }
+
+        @Override
+        public void run() {
+            try{
+                out = new PrintWriter(client.getOutputStream(), true);
+                in = new BufferedReader(new InputStreamReader(client.getInputStream()));
+                out.println("Welcome!\nYou can use the following commands\n" +
```

Comment:
You could use String.format() or a single text block on multiple lines to make the code more readable.